spec/API_specification/statistical_functions.md

```diff
@@ -105,6 +105,15 @@ Calculates the product of input array `x` elements.
 
     -   axis or axes along which products must be computed. By default, the product must be computed over the entire array. If a tuple of integers, products must be computed over multiple axes. Default: `None`.
 
+-   **dtype**: _Optional\[ &lt;dtype&gt; ]_
+
+    -   data type of the returned array. If `None`, the returned array must have the default data type corresponding to the data type "kind" (integer or floating-point) of `x`. If the data type (either specified or resolved) differs from the data type of `x`, the input array should be cast to the specified data type before computing the product. Default: `None`.
```

Comment:
If the default dtype is 32-bit and the input is 64-bit, won't this mean down casting by default?

Reply:
@asmeurer Thanks for pointing this out. Will reword.

Reply:
I've updated the wording to avoid downcasting.